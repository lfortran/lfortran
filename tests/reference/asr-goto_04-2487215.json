{
    "basename": "asr-goto_04-2487215",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/goto_04.f90",
    "infile_hash": "302db59379e85ba1677034f61bec227f237c56085cea27ad7a1554b5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-goto_04-2487215.stdout",
    "stdout_hash": "35e7941446883a7e9e86b18b221c0fab804516bef32e666733e19aa7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}