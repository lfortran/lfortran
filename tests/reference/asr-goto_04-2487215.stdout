(TranslationUnit
    (SymbolTable
        1
        {
            goto_04:
                (Program
                    (SymbolTable
                        2
                        {
                            a:
                                (Variable
                                    2
                                    a
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                ),
                            n:
                                (Variable
                                    2
                                    n
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                )
                        })
                    goto_04
                    []
                    [(Assignment
                        (Var 2 n)
                        (IntegerConstant 2 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerConstant 10 (Integer 4) Decimal)
                        ()
                    )
                    (Select
                        (IntegerBinOp
                            (Var 2 n)
                            Sub
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        [(CaseStmt
                            [(IntegerConstant 1 (Integer 4) Decimal)]
                            [(GoTo
                                1
                                1
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 2 (Integer 4) Decimal)]
                            [(GoTo
                                2
                                2
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 3 (Integer 4) Decimal)]
                            [(GoTo
                                3
                                3
                            )]
                            .false.
                        )]
                        []
                        .false.
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 5 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        1
                        1
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 10 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        2
                        2
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 20 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        3
                        3
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 30 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 a)
                            NotEq
                            (IntegerConstant 70 (Integer 4) Decimal)
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Select
                        (IntegerBinOp
                            (Var 2 n)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        [(CaseStmt
                            [(IntegerConstant 1 (Integer 4) Decimal)]
                            [(GoTo
                                4
                                4
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 2 (Integer 4) Decimal)]
                            [(GoTo
                                5
                                5
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 3 (Integer 4) Decimal)]
                            [(GoTo
                                6
                                6
                            )]
                            .false.
                        )]
                        []
                        .false.
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 5 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        4
                        4
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 40 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        5
                        5
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 50 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        6
                        6
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 60 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 a)
                            NotEq
                            (IntegerConstant 130 (Integer 4) Decimal)
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Select
                        (IntegerBinOp
                            (Var 2 n)
                            Mul
                            (Var 2 n)
                            (Integer 4)
                            ()
                        )
                        [(CaseStmt
                            [(IntegerConstant 1 (Integer 4) Decimal)]
                            [(GoTo
                                7
                                7
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 2 (Integer 4) Decimal)]
                            [(GoTo
                                8
                                8
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(IntegerConstant 3 (Integer 4) Decimal)]
                            [(GoTo
                                9
                                9
                            )]
                            .false.
                        )]
                        []
                        .false.
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 5 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        7
                        7
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 70 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        8
                        8
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 80 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (GoToTarget
                        9
                        9
                    )
                    (Assignment
                        (Var 2 a)
                        (IntegerBinOp
                            (Var 2 a)
                            Add
                            (IntegerConstant 90 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (If
                        (IntegerCompare
                            (Var 2 a)
                            NotEq
                            (IntegerConstant 375 (Integer 4) Decimal)
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
